File: recyclerlayout/src/main/java/com/angel/utils/SWSlipeManager.java
package com.angel.utils;

import com.angel.layout.SWSlipeLayout;

public class SWSlipeManager {

    private SWSlipeLayout swSlipeLayout;
    private static SWSlipeManager SWSlipeManager = new SWSlipeManager();

    public static SWSlipeManager getInstance() {
        return SWSlipeManager;
    }

    public void setSwSlipeLayout(SWSlipeLayout swSlipeLayout) {
        this.swSlipeLayout = swSlipeLayout;
    }

    public void clear() {
        swSlipeLayout = null;
    }

    public void close() {
        if (swSlipeLayout != null) {
            swSlipeLayout.close();
        }
    }

    /**
     * if s==null means no item is open
     *
     * @return ture means open else close
     */
    public boolean haveOpened() {
        return swSlipeLayout != null;
    }

    /**
     * if s==null means no item is open
     *
     * @return true means two item is not the same one and one item is open
     */
    public boolean haveOpened(SWSlipeLayout s) {
        return swSlipeLayout != null && swSlipeLayout == s;
    }
}

File: kotlinlibrary/src/main/java/com/kotlin/utils/SWSlipeManager.kt
package com.kotlin.utils


import com.kotlin.layout.SWSlipeLayout

class SWSlipeManager {

    private var swSlipeLayout: SWSlipeLayout? = null

    fun setSwSlipeLayout(swSlipeLayout: SWSlipeLayout) {
        this.swSlipeLayout = swSlipeLayout
    }

    fun clear() {
        swSlipeLayout = null
    }

    fun close() {
        if (swSlipeLayout != null) {
            swSlipeLayout!!.close()
        }
    }

    /**
     * if s==null means no item is open

     * @return ture means open else close
     */
    fun haveOpened(): Boolean {
        return swSlipeLayout != null
    }

    /**
     * if s==null means no item is open

     * @return true means two item is not the same one and one item is open
     */
    fun haveOpened(s: SWSlipeLayout): Boolean {
        return swSlipeLayout != null && swSlipeLayout == s
    }

    companion object {
        val instance = SWSlipeManager()
    }
}